### PR TITLE
Allow editing a child. 

### LINKED ISSUE #603: Pouvoir modifier un enfant déclaré
Après avoir ajouté un enfant durant sa simulation, il est impossible de le modifier : on peut seulement l'effacer pour en créer un autre.

### PR DESCRIPTION
**Pull request by alexsegura**
Fixes #603. 

I didn't find a better way than adding 2 new routes: 
- `/foyer/enfants/ajouter`
- `/foyer/enfants/:id`

The route changes are now triggering the form display, instead of functions. 
I still need to test some edge cases + DRY. 

mesaides_editchild

### CODE REVIEW

**Comment by alexsegura** [comment]
Voilà, on peut aussi cliquer sur le bloc directement 🐁

mesaides_editchild_click

File: app/js/controllers/foyer.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ angular.module('ddsApp').controller('FoyerCtrl', function($scope, $state, $state
 
     $scope.$on('enfants', function(e, enfants) {
         SituationService.setEnfants(situation, enfants);
-        $state.go('foyer.conjoint');
+        $state.go('foyer.conjoint', {}, { location: 'replace' });
```
Comment (guillett) [inline app/js/controllers/foyer.js]: Pour des `{location: 'replace'}` ?

**Comment by guillett** [comment]
J'ai fait un test et je suis bloqué un écran plus loin.

Cas problématique : Personne isolée avec un enfant à charge.

**Comment by guillett** [comment]
J'ai dit des bêtises, cet après-midi je n'arrive pas à reproduire le problème que j'avais eu (qui était l'impossibilité de passer à l'étape suivante pour un parent isolé). Désolé

**Comment by alexsegura** [comment]
@guillett je pense que j'ai résolu le dernier problème. 

Pour rappel, on avait un problème quand on éditait la date de naissance d'un enfant, le sélecteur "scolarité" était vide, car [on supprime la clé](https://github.com/betagouv/mes-aides-ui/blob/45a9e23c101da377f61da4057244656f81b2719c/app/js/controllers/foyer/individuForm.js#L134) (illustration ci-dessous). 

scolarite

Au lieu de merger un individu déjà existant avec les valeurs par défaut (solution retenue la semaine dernière), j'ai préféré définir une valeur par défaut au niveau du select. 
Je pense que c'est plus logique et moins dangereux. 

À noter que dans 8ee0707349b780444eeca2dd5e7f07136c23049e, j'ai centralisé la recherche de l'individu en cours d'édition dans `individuForm.js`. Ça évite de changer `$scope.individu` dans `FoyerEnfantCtrl`, ce qui était complexe à comprendre. 

⚠️ Du coup, je fais systématiquement un `angular.copy` quand on édite un individu, ce qui signifie que les changements sont reflétés dans le panneau latéral une fois qu'on a validé. 
Est-ce que ça te convient ?

**Comment by guillett** [comment]
C'est vraiment top ton travail @alexsegura. Merci beaucoup !

L'utilisation d'une variable intermédiaire pour la valeur par défaut de certains champs n'est pas pratique car il faut une variable intermédiaire par champ avec une valeur par défaut.

Ton implémentation ne gérait que la valeur par défaut de `scolarite`.`taux_incapacite` et `ass_precondition_remplie` n'avait pas de valeur par défaut. Ces trois champs sont les seuls à être potentiellement rajouté via la recopie de `DEFAULT_INDIVIDU` ce qui ne me parait pas dangereux et éviter lees variables intermédiaires.

J'ai modifié l'initialisation de l'individu pour revenir à la solution proposée la semaine dernière.
